diskqueue.go

```diff
@@ -186,8 +186,8 @@ func (d *diskQueue) start() error {
 		d.logf(ERROR, "DISKQUEUE(%s) failed to retrieveMetaData - %s", d.name, err)
 	}
 
-	fileNameRegexp = regexp.MustCompile(`^` + d.name + `.diskqueue.\d+.dat$`)
-	badFileNameRegexp = regexp.MustCompile(`^` + d.name + `.diskqueue.\d+.dat.bad$`)
+	d.fileNameRegexp = regexp.MustCompile(`^` + d.name + `.diskqueue.\d+.dat$`)
```

Comment:
Though golant regexp is safe for concurent use. i guess there is a small chance in the old shared var that this MustCompile got executed at the same time when another queue is doing a match, which i guess would be problematic.
Instead of converting the regexp to per-instance, another approach is to move the MustCompile to init() so it only execute once. But i'm not familiar with regexp implementation enough to determine how much memory saving it is. this change looks good to me. will see if there's more comment from @kev1n80 @CatherineF-dev

Reply:
regexps might be cached globally if names not unique, but that's the only optimization possible here, otherwise you've got a bug I mentioned in comment